### ISSUE TITLE
(Queue) Push-OutputBinding support initialVisibilityDelay

### ISSUE THREAD

**Issue by spaelling** [Azure/azure-functions-host#6525]
I need to put something into a storage queue with a delay before it will be processed next. Currently resorting to below which is obviously far from optimal. Am I missing something with `Push-OutputBinding` that could do this in a better way?

```powershell
    $Value = @{
        'MeaningOfLife' = 42
    }
    Start-Sleep -Seconds 5
    Push-OutputBinding -Name 'myqueue' -Value $Value   
```

**Comment by AnatoliB** [comment]
This seems to be a feature request for the queue binding in general, not specific to PowerShell.

**Comment by spaelling** [comment]
> 
> 
> This seems to be a feature request for the queue binding in general, not specific to PowerShell.

I may be misunderstanding, but I am talking about the `visibilitytimeout`, see [URI Parameters](https://docs.microsoft.com/en-us/rest/api/storageservices/put-message#uri-parameters) in the Queue service REST API.

> A visibility timeout can also be specified to make the message invisible until the visibility timeout expires.

ie. the message will not trigger a function before the "visibility has timed out".

**Comment by AnatoliB** [comment]
@spaelling Yes, this is how I understand this, thank you. `Push-OutputBinding` does not invoke REST APIs directly. Instead, it communicates with a common framework that handles bindings for all languages in Functions. What you are asking for is a great feature request, but we would need to enable it for queue binding in general, not as a PowerShell-specific capability. This is why I've transferred the issue to a more appropriate repo.

**Comment by f-bader** [comment]
Four years later this seems still to be an unsolved issue. @spaelling did you ever solve it for your usecase?

**Comment by spaelling** [comment]
> Four years later this seems still to be an unsolved issue. @spaelling did you ever solve it for your usecase?

I think I started using Service Bus queues as these have the option to delay processing. This was in Node.js however, so not sure about using Powershell.
